src/test/java/org/opensearch/neuralsearch/query/HybridQueryInnerHitsIT.java

```diff
@@ -0,0 +1,582 @@
+            TEST_MULTI_DOC_WITH_PARENT_CHILD_INDEX_NAME,
+            hybridQueryBuilder,
+            null,
+            10,
+            Map.of("search_pipeline", NORMALIZATION_SEARCH_PIPELINE)
+        );
+
+        List<Object> hitsNestedList = getInnerHitsFromSearchHits(searchResponseAsMap);
+        assertEquals(1, getHitCount(searchResponseAsMap));
+        Map<String, ArrayList<Integer>> innerHitCountPerFieldName = getInnerHitsTotalCountOfNestedField(
+            hitsNestedList,
+            List.of(TEST_PARENT_CHILD_INNER_HITS_FIELD_NAME)
+        );
+        assertEquals(1, innerHitCountPerFieldName.get(TEST_PARENT_CHILD_INNER_HITS_FIELD_NAME).get(0).intValue());
+    }
+
+    @SneakyThrows
+    public void testInnerHits_whenMultipleSubqueriesOnNestedAndParentChildFields_thenSuccessful() {
+        initializeIndexIfNotExist(TEST_MULTI_DOC_WITH_NESTED_PARENT_CHILD_INDEX_NAME);
+        createSearchPipeline(NORMALIZATION_SEARCH_PIPELINE, DEFAULT_NORMALIZATION_METHOD, DEFAULT_COMBINATION_METHOD, Map.of());
+        NestedQueryBuilder nestedQueryBuilder = new NestedQueryBuilder("user", new MatchQueryBuilder("user.name", "John"), ScoreMode.Avg);
+        nestedQueryBuilder.innerHit(new InnerHitBuilder());
+        HybridQueryBuilder hybridQueryBuilder = new HybridQueryBuilder();
+        HasChildQueryBuilder hasChildQueryBuilder = new HasChildQueryBuilder("child", nestedQueryBuilder, ScoreMode.Avg);
+        hasChildQueryBuilder.innerHit(new InnerHitBuilder());
+        hybridQueryBuilder.add(hasChildQueryBuilder);
+        Map<String, Object> searchResponseAsMap = search(
+            TEST_MULTI_DOC_WITH_NESTED_PARENT_CHILD_INDEX_NAME,
+            hybridQueryBuilder,
+            null,
+            10,
+            Map.of("search_pipeline", NORMALIZATION_SEARCH_PIPELINE)
+        );
+
+        List<Object> hitsNestedList = getInnerHitsFromSearchHits(searchResponseAsMap);
+        assertEquals(1, getHitCount(searchResponseAsMap));
+        Map<String, ArrayList<Integer>> innerHitCountPerFieldName = getInnerHitsTotalCountOfNestedField(
+            hitsNestedList,
+            List.of(TEST_PARENT_CHILD_INNER_HITS_FIELD_NAME)
+        );
+        assertEquals(1, innerHitCountPerFieldName.get(TEST_PARENT_CHILD_INNER_HITS_FIELD_NAME).get(0).intValue());
+        Map<String, Object> childInnerHit = (Map<String, Object>) hitsNestedList.get(0);
+        Map<String, Object> childHit = (Map<String, Object>) childInnerHit.get(TEST_PARENT_CHILD_INNER_HITS_FIELD_NAME);
+        List<Object> childInnerHits = getInnerHitsFromSearchHits(childHit);
+        Map<String, ArrayList<Integer>> childInnerHitCountPerFieldName = getInnerHitsTotalCountOfNestedField(
+            childInnerHits,
+            List.of(TEST_NESTED_FIELD_NAME_1)
+        );
+        assertEquals(1, childInnerHitCountPerFieldName.get(TEST_NESTED_FIELD_NAME_1).get(0).intValue());
+    }
+
+    @SneakyThrows
+    public void testInnerHits_withSortingAndPagination_thenSuccessful() {
```

Comment:
Where are we doing pagination in this test case? Also, can we split this test case into 2 one with pagination and other with sorting

Reply:
We are already trying to use resources in a frugal way. So I combined both the test cases. Pagination is applied at inner_hits. Check the inner_hits builder object.